src/helpers.rs

```diff
@@ -518,21 +537,106 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         Ok((true, string_length))
     }
 
+    /// Helper function to write an OsStr as a 0x0000-terminated u16-sequence, which is what
+    /// the Windows APIs usually handle. This function returns `Ok((false, length))` without trying
+    /// to write if `size` is not large enough to fit the contents of `os_string` plus a null
+    /// terminator. It returns `Ok((true, length))` if the writing process was successful. The
+    /// string length returned does not include the null terminator.
+    fn write_os_str_to_wide_str(
+        &mut self,
+        os_str: &OsStr,
+        mplace: MPlaceTy<'tcx, Tag>,
+        size: u64,
+    ) -> InterpResult<'tcx, (bool, u64)> {
+        #[cfg(target_os = "windows")]
```

Comment:
AFAIK one can also do `cfg(windows)`. Do you know what is preferred?

Reply:
I'm not sure which is preferred, but `#[cfg(windows)]` looks more concise and desirable to me.

Reply:
That's also what I used in my WIP patch, so let's consistently use `cfg(windows)` then.